content/en/service_management/app_builder/build.md

```diff
@@ -1,5 +1,216 @@
+
+To delete or duplicate a component, select the component and click the three dot ellipsis (*...*) to display the **Delete** or **Duplicate** options.
+
+Available UI components:
+- Button
+- Callout Value
+- Checkbox
+- Container
+- Date Range Picker
+- JSON Input
+- Modal
+- Number Input
+- Radio
+- Search
+- Select
+- Table
+- Text
+- Text Input
+
+#### Events
+
+UI Components can trigger reactions on an **Event**. The event trigger differs according to the component. For example, a button component can can trigger on a click event, and a table component event can trigger on a page change or table row click.
+
+An event can set a UI component state, open or close a modal, trigger another query, or even run custom JavaScript.
+
+For example, the [GitHub PR Summarizer][4] blueprint uses a **Summarize** button with an event that triggers on a click. The event uses the **Trigger Query** reaction which runs the `summarizePulls` query.
+
+#### Dynamic table values
+
+Similar to [post-query transformation](#post-query-transformation), the table UI component allows you to customize the data source for the table. You can use the **Data Source** field to dynamically fill table values and constrain which objects are pulled into the table as columns.
+
+For example, the [GitHub PR Summarizer][4] blueprint uses a series of GitHub queries to summarize a list of pull requests in a repository. The query uses the data source entry below to constrain the table to 6 columns: `title`,`Summary`,`updated_at`,`user`,`html_url`, and `state`. The highlighted code dynamically populates the user column for each pull request with the author's avatar and GitHub username.
+
+{{< highlight js "hl_lines=17" >}}
+${(() => {
+    const summaryById = Object.fromEntries(
+        summarizePulls.outputs.map(({id, summary}) => [id, summary])
+    );
+    return listPulls.outputs.map(result => {
+        const {title, updated_at, user, state, html_url} = result;
+        const updatedAt = new Date(result.updated_at);
+        let summary;
+        if (summarizePulls.isLoading) {
+            summary = 'Summarizing';
+        } else {
+            summary = summaryById[result.id] ?? 'N/A';
+        }
+        return {
+            title: `**${title}**`,
+            updated_at: updatedAt.toLocaleString(),
+            user: {label: user.login, src: user.avatar_url},
+            summary,
+            state, html_url};
+    })
+})()}
+{{< /highlight >}}
+
+In the table, the **User** column is filled with an avatar and GitHub username for each PR author.
+
+{{< img src="service_management/app_builder/dynamic_tables.png" alt="The GitHub PR Summarizer" style="width:100%;" >}}
+
+### Queries
+
+Queries form the logic behind your app and enable interactions with Datadog integrations. Queries can take inputs from other queries or from UI components, and return outputs for use in other queries or UI components.
```

Comment:
`Queries form the logic behind your app` isn't really accurate. Queries populate your app with data, they aren't really the "logic behind the app". maybe: `Queries allow you to populate your app with data from Datadog APIs or any of our XXX supported integrations`?